src/VizTab.tsx

```diff
@@ -37,23 +129,65 @@ export default function VizTab(props: {namespace: string}) {
     setLocAlertOpen(false);
   }, []));
 
-  const handleAlertClose = (event?: React.SyntheticEvent, reason?: string) => {
-    if (reason !== "clickaway")
-      setLocAlertOpen(false);
-  }
+  const clickCanvas = () => {
+    if (clickAction === "Default") {
+      return;
+    }
+    if (clickAction === "Localize") {
+      const pos = canvasUtils.current?.worldMousePos;
 
-  const [locShowMap, setLocShowMap] = useStorage("Localization.showMap", true);
-  const [scanShow, setScanShow] = useStorage("LaserScan.show", true);
-  const [vizShowPoints, setVizShowPoints] = useStorage("Viz.showPoints", false);
-  const [vizShowLines, setVizShowLines] = useStorage("Viz.showLines", false);
+      if (ws?.connected) {
+        ws.ws?.send(createLocalization2DMsg({
+          namespace: props.namespace,
+          frame: "map",
+          map: rc?.mapName ?? "n/a", // TODO: map selector?
+          x: pos?.x ?? 0,
+          y: pos?.y ?? 0,
+          theta: 0 // TODO: implement drag to set angle
+        }));
+      }
+    } 
+    if (clickAction === "SetNav") {
+      const pos = canvasUtils.current?.worldMousePos;
 
+      if (ws?.connected) { 
+        ws.ws?.send(createPose2DfMsg({
+          namespace: props.namespace,
+          x: pos?.x ?? 0,
+          y: pos?.y ?? 0,
+          theta: 0 // TODO: implement drag to set angle
+        }));
+      }
+    }
+  }
+
+  // settings panel and FAB
   const settingsPanel = <SettingsPanel>
     <List >
       <CollapserItem label="Localization">
-        <FormControlLabel
-          control={<Switch checked={locShowMap} onClick={() => setLocShowMap(s => !s)}/>}
-          label="Show map"
-        />
+        <FormGroup row>
+          <FormControlLabel
+            control={<Switch checked={locShowMap} onClick={() => setLocShowMap(s => !s)}/>}
+            label="Show map"
+          />
+        </FormGroup>
+        <FormGroup row>
+          <Button 
+            onClick={() => setClickAction(a => (a === "Localize") ? "Default" : "Localize")}
+            variant="outlined"
+            color={clickAction === 'Localize' ? 'primary' : 'default'}
+          >
+            Set Pose
+          </Button>
+          <Box flexGrow="1" />
+          <Button 
+            onClick={() => setClickAction(a => (a === "SetNav") ? "Default" : "SetNav")}
+            variant="outlined"
+            color={clickAction === 'SetNav' ? 'primary' : 'default'}
+          >
+            Set NavGoal
+          </Button>
+        </FormGroup>
       </CollapserItem>
```

Comment:
future: could create a map dropdown here. Maybe you need a separate `selectedMapName` state since the robot will likely just reset it before the user sends a localize command.